lib/mechanize/http/agent.rb

```diff
@@ -1234,7 +1234,7 @@ def set_proxy addr, port, user = nil, pass = nil
       end
     end
 
-    proxy_uri = URI "http://#{addr}"
+    proxy_uri = URI (addr =~ %r(^https?://) ? addr : "http://#{addr}")
```

Comment:
Can you make this two lines?

Reply:
I've updated the commit to split the string manipulation from the URI parsing. Please let me know if there's anything else, as I'm relatively fresh to this.